File: Organizing Data Java/src/tdtu/edu/OrganizingData/EncapsulateCollection/Organizing/Library.java
package tdtu.edu.OrganizingData.EncapsulateCollection.Organizing;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

import tdtu.edu.OrganizingData.EncapsulateCollection.Book;

public class Library {
	Set<Book> books =null;

	public Library() {
		super();
		books= new HashSet<Book>();	
	}

	public Set<Book> getBooks() {
		
		return Collections.unmodifiableSet(books);
	}

	public void addBooks(Book book) {
		this.books.add(book);
	}
	
	public void removeBooks(Book book) {
		this.books.remove(book);
	}
	
}

File: Organizing Data C#/DesignPatternGK/ConsoleApp1/Library.cs
using ConsoleApp1;
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EncapsulateCollection
{
    class Library
    {
        private HashSet<Book> _books = null;

        public HashSet<Book> Books{
            get { return _books; }
        }

        public Book Book
        {
            set { _books.Add(value); }
        }

        public void remove(Book book)
        {
            _books.Remove(book);
        }

        public Library()
        {
            _books = new HashSet<Book>();
        }

        
    }
}
